src/routes/quickstarts/cypress/setup.mdx

```diff
@@ -0,0 +1,180 @@
+import { Tabs, Tab } from "../../../components/tabs"
+import Snippet from "../../../components/snippet"
+
+# Mock Network Requests in Cypress with Mirage
+
+Use your Mirage server to test your application under different server scenarios using [Cypress](https://www.cypress.io/).
+
+> This is a quickstart guide for people already using `cypress` in their apps.
+
+## Step 1: Install Mirage
+
+First, make sure you have Mirage installed:
+
+```bash
+# Using npm
+npm install --save-dev miragejs
+
+# Using Yarn
+yarn add --dev miragejs
+```
+
+## Step 2: Define your server
+
+Create a new `src/server.js` file and define your mock server.
+
+Here's a basic example:
+
+```js
+// src/server.js
+import { Server, Model } from "miragejs"
+
+export function makeServer({ environment = "development" } = {}) {
+  let server = new Server({
+    environment,
+
+    models: {
+      user: Model,
+    },
+
+    seeds(server) {
+      server.create("user", { name: "Bob" })
+      server.create("user", { name: "Alice" })
+    },
+
+    routes() {
+      this.namespace = "api"
+
+      this.get("/users", schema => {
+        return schema.users.all()
+      })
+    },
+  })
+
+  return server
+}
+```
+
+## Step 3: Add the Mirage Cypress configuration
+
+Add the following Mirage bootstrapping code and command to your `cypress/support/index.js` file:
+
+```js
+// cypress/support/index.js
+Cypress.on("window:before:load", win => {
+  win.handleFromCypress = function(request) {
+    return fetch(request.url, {
+      method: request.method,
+      body: request.requestBody,
+    }).then(res => {
+      if (res.headers.map["content-type"] === "application/json") {
+        return res.json()
+      } else {
+        return ""
+      }
+    })
+  }
+})
+```
+
+## Step 4: Start Mirage when Cypress is enabled
+
+In your app's bootstrapping file, start Mirage if `window.Cypress` is true.
+
+> Create React App users, this code goes in `src/index.js`
+
+> Vue CLI users, this code goes in `src/main.js`.
+
+```js
+import { Server } from "miragejs"
+
+if (window.Cypress) {
+  let cyInterceptor = new Server({
+    routes() {
+      let methods = ["get", "put", "patch", "post", "delete"]
+      methods.forEach(method => {
+        this[method]("/*", (schema, request) => {
+          return window.handleFromCypress(request)
+        })
+      })
+    },
+  })
+  cyInterceptor.logging = false
+}
+```
+
```

Comment:
Let's add just a bit of color here after they see this code, because it's confusing to see miragejs import here. Something like:

"This code is actually setting up a second Mirage server, but its sole purpose is to act as a proxy for our application's network requests. This is because our real configured Mirage server is going to be running alongside our Cypress code (so that our tests have access to it), and our Cypress code runs in a completely separate environment from our application."

Reply:
I tweaked this a little, but I'm not sure I like it because it feels like it exposes a lot of the inner workings. Thoughts?

This code is setting up a second Mirage server, but its sole purpose is to act as a proxy for our application's network requests. This is because our real configured Mirage server is going to be running alongside our Cypress code, so that our tests have access to it.

The two Mirage servers are linked through the `handleFromCypress` function, which allows our test code to mock out network requests coming from the application's window.

Reply:
Agree this is a lot, how about:

Now, whenever Cypress boots up your app, this code will delegate your app's network requests to the `handleFromCypress` function that we defined in the previous step.

Once we start our real configured Mirage server alongside our Cypress code, it will start intercepting the requests from that function.